### ISSUE TITLE
can not run pom.rb build on Java 11

### ISSUE THREAD

**Issue by kares** [jruby/mavengem#1]
very likely due having old jruby-openssl (jruby): https://github.com/jruby/jruby-openssl/issues/142

```
Caused by: org.jruby.embed.EvalFailedException: (LoadError) load error: jopenssl/load -- java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
	at org.jruby.embed.internal.EmbedEvalUnitImpl.run(EmbedEvalUnitImpl.java:131)
	at org.jruby.embed.ScriptingContainer.runUnit(ScriptingContainer.java:1307)
	at org.jruby.embed.ScriptingContainer.runScriptlet(ScriptingContainer.java:1300)
	at org.torquebox.mojo.rubygems.DefaultRubygemsGateway.<init>(DefaultRubygemsGateway.java:36)
	at org.torquebox.mojo.mavengem.Rubygems.<clinit>(Rubygems.java:27)
	... 53 more
Caused by: org.jruby.exceptions.RaiseException: (LoadError) load error: jopenssl/load -- java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
	at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55)
	at RUBY.<main>(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/openssl.rb:1)
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
	at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55)
	at RUBY.<main>(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/security.rb:1)
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
	at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55)
	at RUBY.<main>(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/security.rb:12)
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
	at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55)
	at RUBY.<main>(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/package.rb:1)
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
	at RUBY.(root)(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/package.rb:44)
        at RUBY.<main>(uri:classloader:/nexus/gemspec_helper_impl.rb:1)
```

for whatever reason I am getting it on 9.1.17.0 as well but only as I run `rmvn` (`mvn` seems fine)

same issue as https://github.com/torquebox/jruby-maven-plugins/issues/91 but I felt its in the wrong place

there's also the unsupported Java version warning, likely coming from maven itself.

```
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building JRuby Lib Setup 9.2.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading: mavengem:https://rubygems.org/rubygems/jar-dependencies/0.4.0/jar-dependencies-0.4.0.pom
unsupported Java version "11", defaulting to 1.7

```

**full-trace** at https://gist.github.com/kares/14c75cd6642016b787337440cd9fb1ad

... guess we'll need to update ruby-maven-libs with newer JRuby and Maven itself?

**Comment by kares** [comment]
also here's an attempted build of JRuby on CI: https://github.com/jruby/jruby/pull/5340

**Comment by kares** [comment]
@mkristian yeah thought so - since I noticed 9.1.2 in the output being used anyway ...
am not sure about 9.2.0 ... would try going for 9.1.17.0 atm - not sure if there are any issues but at least the jossl part should be working (unless we discover smt else still broken in 0.10.1). 
let me know if you need helping around, will be watching the changes you'll be making to learn something about the whole maven plugin setup.

**Comment by mkristian** [comment]
@kares I do have a workaround #2 
BUT I do not understand why it fails. the PR still prints the JRUBY_VERSION just before requiring openssl and it is 9.1.17.0 and it fails with this StringIndexOutOfBoundException
I do not understand the failure. the fix is legit as we really do not need openssl to unpack gem and rubygems/package.rb has switches everywhere to deal with no openssl.

still I want to understand why this fails with jruby-9.1.17.0 - any idea ?

**Comment by kares** [comment]
interesting, to recap - due relying on maven to do the download RG's downloading parts won't be used ...
... which means the real OpenSSL bits aren't needed, can it cause some different issues later on?

> it is 9.1.17.0 and it fails with this StringIndexOutOfBoundException
I do not understand the failure. the fix is legit as we really do not need openssl to unpack gem and rubygems/package.rb has switches everywhere to deal with no openssl.

weird - is it possible that a jossl version is declared somewhere? 
wasn't able to find out how jruby gets packed - there's <jruby.version>, assuming its using complete .jar
in which case it should have worked already ...

**Comment by mkristian** [comment]
@kares yes and actual download happens here: https://github.com/torquebox/mavengem/blob/cb2bf02afcccd61e5f140a108a531930dba54a61/rubygems-tools/src/main/java/org/torquebox/mojo/rubygems/layout/CachingProxyStorage.java#L108
i.e. only java is doing the actual url connection.

any ways going to release this asap

thanks for you input !

**Comment by headius** [comment]
I believe this will be fixed by #9 since that updates JRuby to 9.4.3.
